docs/architecture.md

```diff
@@ -10,19 +10,22 @@ events related to specific kernel subsystems.
 
 ![different tools offered by Inspektor Gadget](images/architecture/ebpf.svg)
 
-On Kubernetes, Inspektor Gadget is deployed to each node as a privileged
-DaemonSet.
-It uses in-kernel eBPF helper programs to monitor events from userspace programs
-in a pod. The eBPF programs are run by the kernel and gather the log data.
-Inspektor Gadget's userspace utilities fetch the log data from ring buffers
-and display it. What eBPF programs are and how Inspektor Gadget uses them is
-briefly explained here.
+Before using Inspektor Gadget, it first need to be deployed to each node as a
+privileged `DaemonSet`.
```

Comment:
I think we should not tell this as a "story" given that this document is not a guide. 

```suggestion
On Kubernetes, Inspektor Gadget is deployed to each node as a DaemonSet.
```

It can be extended with a bit more of information of how it's deployed:

"It's usually deployed by the kubectl gadget plugin. It interacts with the kube-api server to create all the resources needed for Inspektor Gadget to work. Another way of deploying Inspektor Gadget is to directly apply the [manifests](pkg/resources/manifests/deploy.yaml) using kubectl".